### PR TITLE
Add HaikuOS as platform

### PR DESCRIPTION
**Pull request by xspager**
Hi! Is it possible to add [Haiku](https://www.haiku-os.org/) as a supported platform? It is Posix compliant but need some platform specific options like linking with lnetwork in the case of luasocket.

### CODE REVIEW

File: src/luarocks/cfg.lua [hunk 1]
```diff
@@ -140,8 +143,8 @@ local platform_order = {
    -- Windows
    win32 = 10,
    mingw32 = 11,
-   windows = 12 }
-
+   windows = 12, 
+   haiku = 13 }
```
Comment (hishamhm) [inline src/luarocks/cfg.lua]: Please put haiku after cygwin as 10 and increment the ones below it. These numbers don't need to be the same (only their order matters) but I guess it looks nicer if we keep all the Unixes and Windows together. Thanks!

**Comment by xspager** [comment]
Done!

**Comment by hishamhm** [comment]
Thanks!

**Comment by xspager** [comment]
Awesome! Thanks!

**Pull request merged**